### PR TITLE
more error logging for random transaction stream testing

### PR DESCRIPTION
**Pull request by S11001001**
Improving the random transaction stream test from #6658 with additional output about the parameters that led to the error. This might not be enough, as the distribution may play a role.

If an error or cause can't be quickly found, we may reduce the strength of the assertion about `readOffsets` to exclude order problems, and leave an issue for solving later.

From a discussion with @cocreature , @SamirTalwar-DA , and @remyhaemmerle-da .

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [x] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [x] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: build.sh [hunk 1]
```diff
@@ -61,7 +61,8 @@ stop_postgresql # in case it's running from a previous build
 start_postgresql
 
 # Run the tests.
-bazel test //... \
+bazel test //ledger/participant-integration-api:participant-integration-api-tests_test_suite_src_test_suite_scala_platform_store_dao_JdbcLedgerDaoPostgresqlSpec.scala \
+  --test_arg=-z --test_arg="fall back to limit-based query with consistent results" \
```
Comment (SamirTalwar-DA) [inline build.sh:65]: You need to make this change in _build.ps1_ too.

Comment (SamirTalwar-DA) [inline build.sh:65]: You need to make this change in _build.ps1_ too.

Comment (cocreature) [reply build.sh:65]: why? we’re not going to merge this, we just want to see the error.

Comment (SamirTalwar-DA) [reply build.sh:65]: So you can see the error on Windows without other random timeouts. 🙃

Comment (cocreature) [reply build.sh:65]: We saw this error on macos and Linux, I would go as far as just disabling the Windows job for testing here.

**Review by SamirTalwar-DA — APPROVED** [review]

**Pull request merged**